rosidl_runtime_c/include/rosidl_runtime_c/u16string_functions.h

```diff
@@ -31,53 +31,156 @@ extern "C"
  * Size represents the size of the contents of the string, while capacity represents the overall
  * storage of the string (counting the null terminator).
  * All strings must be null-terminated.
+ * The rosidl_runtime_c__U16String structure should be deallocated using the given function
+ * rosidl_runtime_c__U16String__fini() when it is no longer needed.
+ *
+ * \param[inout] str a pointer to a U16 string structure
+ * \return true if successful, false if the passed string pointer is null
+ *   or the memory allocation failed
  */
 ROSIDL_GENERATOR_C_PUBLIC
 bool
 rosidl_runtime_c__U16String__init(rosidl_runtime_c__U16String * str);
 
+/// Deallocate the memory of the rosidl_runtime_c__U16String structure.
+/*
+* Calling the function with an already deallocated sequence is a no-op.
+*
+* \param[inout] str a pointer to a U16 string structure to be finalized
+*/
 ROSIDL_GENERATOR_C_PUBLIC
 void
 rosidl_runtime_c__U16String__fini(rosidl_runtime_c__U16String * str);
 
+/// Assign the uint16_t value of n characters to the rosidl_runtime_c__U16String structure.
+/*
+ *
+ * This function returns `false` if memory cannot be allocated,
+ * if the input uint16_t pointer is null or if the size is higher than SIZE_MAX.
+ * In both cases no error message is set.
+ *
+ * \param[inout] str a pointer to a U16 string structure
+ * \param[in] value uint16_t pointer to be assigned
+ * \param[in] n size of the value string
+ * \return true if successful, false if the passed string pointer is null
+ *   or if the passed value pointer is null or if the size is higher than SIZE_MAX
+ *   or if the memory reallocation failed.
+ */
 ROSIDL_GENERATOR_C_PUBLIC
 bool
 rosidl_runtime_c__U16String__assignn(
   rosidl_runtime_c__U16String * str, const uint16_t * value, size_t n);
 
+/// Assign the c string pointer value of n characters to the rosidl_runtime_c__U16String structure.
+/*
+ *
+ * This function is identical to rosidl_runtime_c__U16String__assignn() except the type of the
+ * value is c string pointer
+ *
+ * \see rosidl_runtime_c__U16String__assignn()
+ *
+ * \param[inout] str a pointer to a U16 string structure
+ * \param[in] value c string pointer to be assigned
+ * \param[in] n size of the value string
+ * \return true if successful, false if the passed string pointer is null
+ *   or if the passed value pointer is null or if the size is higher than SIZE_MAX
+ *   or if the memory reallocation failed.
+ */
 ROSIDL_GENERATOR_C_PUBLIC
 bool
 rosidl_runtime_c__U16String__assignn_from_char(
   rosidl_runtime_c__U16String * str, const char * value, size_t n);
 
+/// Assign the uint16_t pointer to the rosidl_runtime_c__U16String structure.
+/*
+ *
+ * This function is identical to rosidl_runtime_c__U16String__assignn() except the length of the
+ * uint16_t does not have to be given.
```

Comment:
It might be good to mention that `rosidl_runtime_c__U16String__len` is being used to determine the length of the passed string.